src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -437,8 +437,10 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     //"this" is readonly in members of readonly structs, unless we are in a constructor.
                     if (!thisref.Type.IsValueType ||
                             (RequiresAssignableVariable(valueKind) &&
+#pragma warning disable IDE0051 // Formatting document
                              thisref.Type.IsReadOnly && 
                              (this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.Constructor))
+#pragma warning restore IDE0051 // Formatting document
```

Comment:
:bug: Trailing whitespace followed by unchanged text on the next line should only report the diagnostic at the location of the trailing whitespace.

Reply:
right now, removing trailing whitespace is built in engine behavior. no option or rule about it. but we can introduce one if we want to. this behavior is something we inherit from original C# formatter from I don't know Whidbey?

Reply:
In the first iteration, this analyzer reports all differences which would occur if you run Format Document without code cleanup. In some situations, this will include trailing whitespace removal (which is not configurable).

Reply:
> but we can introduce one if we want to.

I'm not convinced this is necessary or desirable. Elimination of trailing whitespace is *almost* universally seen as desirable, and the only blocking issue today is lack of compile-time enforcement. Once the analyzer is available, I expect teams will quickly resolve these issues by applying a one-time formatting operation that removes the spaces, and then continue to use the analyzer as a tool to prevent their reintroduction.

Reply:
sure. I am just saying I think it might be quite easy or at least not to hard to add option if we want to.